java/openjdk8-powerpc/Portfile

```diff
@@ -0,0 +1,324 @@
+    configure.args-delete   --with-extra-cflags="${configure.cflags}" \
+                            --with-extra-cxxflags="${configure.cxxflags}" \
+                            --with-extra-ldflags="${configure.ldflags}"
+}
+
+variant zeroshark \
+    conflicts core zero server \
+    description {JVM with no assembler, zero interpreter and shark/llvm compiler backend} {
+    configure.args-append   --with-jvm-variants=zeroshark \
+                            --with-extra-cflags="${configure.cflags} -I${prefix}/include -arch ${configure.build_arch}" \
+                            --with-extra-cxxflags="${configure.cxxflags} -arch ${configure.build_arch}" \
+                            --with-extra-ldflags="${configure.ldflags} -I${prefix}/lib -arch ${configure.build_arch}"
+    configure.args-delete   --with-extra-cflags="${configure.cflags}" \
+                            --with-extra-cxxflags="${configure.cxxflags}" \
+                            --with-extra-ldflags="${configure.ldflags}"
+}
+
+if {![variant_isset debug] && ![variant_isset release]} {
+    default_variants-append +release
+}
+
+default_variants-append     +zero
+
+build.type          gnu
+build.target        images
+
+use_parallel_build  no
+
+set jdkn jdk1.${major}.0_${update}.jdk
+set jren jre1.${major}.0_${update}.jre
+set jdk_path ${tpath}/JavaVirtualMachines/${real_name}
+set jre_path ${tpath}/JavaVirtualMachines/${real_name}-jre
+
+if {[variant_isset zero]} {
+    # We do not build macOS-style bundles here, so paths differ:
+    set jdk_bundle_dir build/openjdk8/images/j2sdk-image
+    set jre_bundle_dir build/openjdk8/images/j2re-image
+
+    # But we want to install in a macOS-style:
+    destroot {
+        xinstall -m 755 -d ${destroot}${jdk_path}/Contents
+        xinstall -m 755 -d ${destroot}${jre_path}/Contents
+        # This renames j2*-image dirs to respective Homes:
+        move ${worksrcpath}/${jdk_bundle_dir} ${destroot}${jdk_path}/Contents/Home
+        move ${worksrcpath}/${jre_bundle_dir} ${destroot}${jre_path}/Contents/Home
+        # We need these:
+        system "ln -s ${prefix}/lib/libfontconfig.1.dylib ${destroot}${jdk_path}/Contents/Home/jre/lib/libfontconfig.dylib"
+        system "ln -s ${prefix}/lib/libfontconfig.1.dylib ${destroot}${jre_path}/Contents/Home/lib/libfontconfig.dylib"
+    }
+    # Binaries are gone from worksrcpath, so run from destroot:
+    test.cmd        ${destroot}${jdk_path}/Contents/Home/bin/java
+
+} else {
+    set jdk_bundle_dir build/openjdk8/images/j2sdk-bundle/${jdkn}/Contents
+    set jre_bundle_dir build/openjdk8/images/j2re-bundle/${jren}/Contents
+
+    destroot {
+        xinstall -m 755 -d ${destroot}${jdk_path}
+        xinstall -m 755 -d ${destroot}${jre_path}
+        copy ${worksrcpath}/${jdk_bundle_dir} ${destroot}${jdk_path}
+        copy ${worksrcpath}/${jre_bundle_dir} ${destroot}${jre_path}
+    }
+
+    test.cmd        ${worksrcpath}/${jdk_bundle_dir}/Home/bin/java
+}
+
+destroot.violate_mtree  yes
```

Comment:
Is this actually necessary for the port?

Reply:
Most `openjdk*` ports install under `${prefix}/Library/Java/JavaVirtualMachines`, but still use `destroy.violate_mtree` to create a symlink under `/Library/Java/JavaVirtualMachines`, which is where macOS Java tools look for Java installations.

@barracuda156 It might be nice to follow theis pattern for this port as well.

Reply:
@pmetzger The overall design of the port was mostly retained from openjdk8, provided it did not break anything. Destroot was modified in a way to actually work, but I think final locations are identical.

If installation (sub)prefix has to be changed, perhaps it should be done for both ports and in a separate PR.

@breun I do not have particular opinion what is preferable, for MacPorts whichever should work as long as the path is set for dependents.

(Please mind that any random change which modifies installation would mean another ~5 hours of compilation, so I rather avoid changing something without a real need for cosmetic reasons. I have already built this thing at least 4 times since my original PR.)

Reply:
Back when I created new `openjdk*` ports there was a preference for putting as many files as possible under `${prefix}`, and only having the symlink under `/Library/Java/JavaVirtualMachines` outside the MacPorts tree.

The `openjdk8` port hasn't been updated in quite a while (bumping the version didn't work) and it hasn't been aligned with the other `openjdk*` ports in this regard.

Reply:
@breun I will take a look at what jdk11 does. In principle you think we rather do the same for jdk8?
I am not gonna handle Intel case though. Someone will need to deal with that, otherwise we have two different install modes with Intel vs PPC ports.

Reply:
@breun In fact if you can update installation for openjdk8 (i.e. non-ppc one), I can just adapt it to the zero case which ppc uses.

Reply:
I don’t believe the maintainer of `openjdk8`, `openjdk11` and `openjdk17` has been active in quite a while. I have been bumping these ports, but for `openjdk8` just bumping the version beyond 8u372 fails. There’s an open Trac ticket for this and others have been working on it, but no result yet.

Reply:
I don’t think it’s worth bothering users with an update of `openjdk8` that just changes where the files get installed. I would want to combine that with an actual version update (but I don’t know how to fix the failing build for newer versions).

You can look at for instance `openjdk11` for an example that installs its files under `${prefix}` and only creates a symlink outside it.

Reply:
Ok, fair enough.

BTW, does openjdk11 build with openjdk8 (on arm64 or x86)? There is no port for openjdk9, so in a case we consider going up from 8, how that is done?

Reply:
`openjdk11` is currently built using the [openjdk11-bootstrap](https://github.com/macports/macports-ports/blob/master/java/openjdk11-bootstrap/Portfile) port, which installs a prebuilt Oracle OpenJDK 11 on x86_64 and Azul Zulu OpenJDK 11 on arm64. I don't know if there are prebuilt ppc binaries available for OpenJDK 9+, but if they're not, but it's possible to build them yourself, then you should be able to build JDK 9 using a JDK 8 boot JDK, and then build JDK 10 using a JDK 9 boot JDK, etc.

Reply:
@breun Got it, thanks. This is not in any near plans, I rather try fixing X11 in jdk8 first. If that works, then it could make sense to build something later.